### PR TITLE
Add validation status field to the sync tables

### PR DESCRIPTION
**Pull request by keymane**

### CODE REVIEW

File: opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java [hunk 1]
```diff
@@ -259,69 +241,76 @@ public void insert(SQLiteDatabase db, Event event, JSONObject serverJsonObject)
...
-                        lastServerVersion = c.getServerVersion();
+        try {
+            long lastServerVersion = 0l;
+
+            getWritableDatabase().beginTransaction();
+
+            for (int i = 0; i < array.length(); i++) {
+                Object o = array.get(i);
+                if (o instanceof JSONObject) {
+                    JSONObject jo = (JSONObject) o;
```
Comment (pld) [inline opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java]: If o is an instance of JSONObject why is this type conversion necessary?

File: opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java [hunk 2]
```diff
@@ -259,69 +241,76 @@ public void insert(SQLiteDatabase db, Event event, JSONObject serverJsonObject)
...
-            if (o instanceof JSONObject) {
-                JSONObject jo = (JSONObject) o;
-                Event e = convert(jo, Event.class);
-                if (e != null) {
-                    insert(getWritableDatabase(), e, jo);
-                    if (e.getServerVersion() > 0l) {
-                        lastServerVersion = e.getServerVersion();
+            getWritableDatabase().beginTransaction();
+
+            for (int i = 0; i < array.length(); i++) {
```
Comment (pld) [inline opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java]: Isn't this the same code as around line 259? Is it possible to DRY it up

File: opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java (hunk 1 above)
Comment (keymane) [reply opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java]: o is an Object in line 260 so we need to cast it to a JSONObject

**Comment by pld** [comment]
future todo I'd say we change all `InValid` to `Invalid`, it seems weird to capitalize the v, Invalid is a single word, not a conjunction of two words

**Pull request merged**